### ISSUE TITLE
[Feature Request] - Allow Variable Replacement on Visual Studio Test Plans

### ISSUE THREAD

**Issue by MattDavisRM** [Microsoft/azure-pipelines-tasks#9725]
I am using a release pipeline that has been configured and contains a Visual Studio Test task (v2.*).  I am wanting to spawn the release pipeline using the REST API.  It is possible that I'd like to pass the Test Plan and Test Suite as a variable.  The task though does not allow variable substitution for Test suite.  It might allow it for the others, but I haven't actually run the pipeline to confirm (it lets me freeform enter).

The workaround I'm leveraging now is to create a draft release, then update the steps configuration manually.  I'd prefer not to have to do the draft release because it can be a little brittle or a lot of processing to capture the right step for the Test step.

**Is it possible that Variable subsitution would be considered for the Test Plan / Test Suite selection when using the Test plan test selection?**

**Comment by acesiddhu** [comment]
@MattDavisRM what is the scenario? this is not supported today.

**Comment by MattDavisRM** [comment]
@acesiddhu We have built a custom deployment front-end for users that models a selection for Environment, Build, Deploy settings, and optionally any test execution that they want to execute after deployment.  We have created a release pipeline per environment that handles the deployment and can start the tests.  We are using the API calls to create the release and start the release.  From the selection that is available to users, they can select the test suite/test plan that they want executed.  In the pipeline though, the test plan/test suite selection will only allow a pre-defined value, so when executing a release pipeline, there is no way to dynamically use a variable to choose which plan should be executed.  We'd just like a way for us to pass a variable to the pipeline (the IDs for the test suite/test plan) and have that be replaced in what the test will execute.

As it stands now, my plan is to create a draft release, get the release details, then directly update the workflowTask inputs for test suite/test plan, then do a POST back to update the release.
